### PR TITLE
Bump ERC1967Utils pragma to 0.8.21 and set Slither CI version to 0.10.1

### LINKED ISSUE #4995: Incorrect pragma
PR #4987 uses qualified access to events from other contracts. This was introduced in 0.8.21. Yet `proxy/utils/ERC1967Utils.sol` uses `pragma solidity ^0.8.20`

This leaves us with two options:
- revert #4987
- increase the pragma to `^0.8.21`

### PR DESCRIPTION
**Pull request by dimitriospapathanasiou**
Fixes #4995  

I noticed the related issue about the pragma that is used. After minor local tests, I verified that the increase to 0.8.21 works well.

Same as #4999 but with the proper files added. Excuse my previous PR, I am a first time contributor. Sorry for any inconvenience.

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [x] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Review by Amxx — DISMISSED** [review]
Thank you @dimitriospapathanasiou 

We don't need a changeset for that.

We need @ernestognw 's 2nd approval and we are good.

**Comment by ernestognw** [comment]
Looks good to me as well, I just added a note to the changelog referring to the breaking changes arising from this.

**Review by ernestognw — APPROVED** [review]

**Comment by ernestognw** [comment]
CI is broken due to Slither 0.10.2. The rule breaking is [pragma](https://github.com/crytic/slither/wiki/Detector-Documentation#different-pragma-directives-are-used) but I'm struggling to see why it's failing now.

Let's not unblock this and set the Slither version to 0.10.1 for now.

**Review by ernestognw — APPROVED** [review]

**Pull request merged**
